cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -996,6 +996,16 @@ TEST_F(TestTakeKernel, InvalidIndexType) {
                                          "[0.0, 1.0, 0.1]", &arr));
 }
 
+TEST_F(TestTakeKernel, TakeEmptyIndices) {
+  Datum out;
+  auto dat = Datum(std::make_shared<ChunkedArray>(ArrayFromJSON(int8(), "[]")));
+  auto idx = Datum(std::make_shared<ChunkedArray>(ArrayVector(), int32()));
+  auto ret = Take(dat, idx).Value(&out);
+  ASSERT_OK(ret);
```

Comment:
```suggestion
TEST_F(TestTakeKernel, TakeEmptyIndices) {
  Datum dat = ChunkedArrayFromJSON(int8(), {"[]"});
  Datum idx = ChunkedArrayFromJSON(int32(), {});
  ASSERT_OK_AND_ASSIGN(auto out, Take(dat, idx));
```

Reply:
This can be simplified a bit.

Reply:
I think we should test both branches above too. This tests the changes in TestCC, we can add a test for TestAC that calls Take with an Array and an empty (chunk-less) ChunkedArray.

Reply:
Thanks David!